aeppl/transforms.py

```diff
@@ -89,52 +91,115 @@ def transformed_logprob(op, value, *inputs, name=None, **kwargs):
+DEFAULT_TRANSFORM = DefaultTransformSentinel()
 
-    if rv_map_feature is None:
-        return None  # pragma: no cover
 
-    if not isinstance(node.op, RandomVariable):
-        return None  # pragma: no cover
+@local_optimizer(tracks=None)
+def transform_values(fgraph: FunctionGraph, node: Node) -> Optional[List[Node]]:
+    """Apply transforms to value variables.
 
-    trans_node = _default_transformed_rv(node.op, node)
+    It is assumed that the input value variables correspond to forward
+    transformations, usually chosen in such a way that the values are
+    unconstrained on the real line.
 
-    if trans_node is None:
-        return None  # pragma: no cover
+    For example, if ``Y = halfnormal(...)``, we assume the respective value
+    variable is specified on the log scale and back-transform it to obtain
+    ``Y`` on the natural scale.
+    """
 
-    # Get the old value variable and remove it from our value variables map
-    rv_var = node.outputs[1]
-    old_value_var = rv_map_feature.rv_values.pop(rv_var)
+    rv_map_feature = getattr(fgraph, "preserve_rv_mappings", None)
+    values_to_transforms = getattr(fgraph, "values_to_transforms", None)
+
+    if rv_map_feature is None or values_to_transforms is None:
+        return None  # pragma: no cover
 
-    transform = trans_node.op.transform
+    try:
+        rv_var = node.default_output()
+    except ValueError:
+        return None
+
+    value_var = rv_map_feature.rv_values.get(rv_var, None)
+    if value_var is None:
+        return None
+
+    transform = values_to_transforms.get(value_var, None)
+
+    if transform is None:
+        return None
+    elif transform is DEFAULT_TRANSFORM:
+        trans_node = _default_transformed_rv(node.op, node)
+        if trans_node is None:
+            return None
+        transform = trans_node.op.transform
+    else:
+        new_op = _create_transformed_rv_op(node.op, transform)()
+        trans_node = new_op.make_node(*node.inputs)
+        trans_node.outputs[1].name = node.outputs[1].name
 
     # We now assume that the old value variable represents the *transformed space*.
     # This means that we need to replace all instance of the old value variable
     # with "inversely/un-" transformed versions of itself.
-    new_value_var = transform.backward(old_value_var, *trans_node.inputs)
-    rv_map_feature.rv_values[rv_var] = new_value_var
+    new_value_var = transform.backward(value_var, *trans_node.inputs)
+    if value_var.name and getattr(transform, "name", None):
+        new_value_var.name = f"{value_var.name}_{transform.name}"
 
+    # Map both new and old rv to new value var
     new_rv_var = trans_node.outputs[1]
-
-    if old_value_var.name and getattr(transform, "name", None):
-        new_value_var.name = f"{old_value_var.name}_{transform.name}"
-
     rv_map_feature.rv_values[new_rv_var] = new_value_var
+    rv_map_feature.rv_values[rv_var] = new_value_var
```

Comment:
Do we need this association?

Reply:
It was there originally. I am not sure we do though